bottlecap/src/logs/agent.rs

```diff
@@ -22,14 +22,15 @@ pub struct LogsAgent {
 impl LogsAgent {
     #[must_use]
     pub fn run(
-        function_arn: &str,
-        tags_provider: Arc<provider::Provider>,
+        tags_provider: Arc<tags::provider::Provider>,
         datadog_config: Arc<config::Config>,
     ) -> LogsAgent {
-        let function_arn = function_arn.to_string();
         let aggregator: Arc<Mutex<Aggregator>> = Arc::new(Mutex::new(Aggregator::default()));
-        let mut processor: LambdaProcessor =
-            LambdaProcessor::new(function_arn, tags_provider, Arc::clone(&datadog_config));
+        let mut processor = processor::ProcessorType::new(
```

Comment:
nit: the name for a module being `processor::ProcessorType` is a little strange. I'd expect that for an enum declaration but not a concrete value. Clippy might complain about using the file name in the module name (#pedantic).

What if we used `LogProcessor`?

Reply:
Going forward with `LogsProcessor` since I don't know if `Provider` is ideal, yet I'm still not sure if `LogsProcessor` would be good for the future either 😕